File: java/playwright/src/test/java/dev/selenium/request/AbortTest.java
package dev.selenium.request;

import com.microsoft.playwright.Route;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class AbortTest extends TestBase {

  @Test
  public void unconditionalAbort() {
    page.route("**/*.css", Route::abort);

    page.navigate("https://selenium.dev");
    double height = page.locator("svg").first().boundingBox().height;
    Assertions.assertTrue(height > 30);
  }

  @Test
  public void conditionalAbort() {
    page.route(
        "**/*",
        r -> {
          if ("image".equals(r.request().resourceType())) {
            r.abort();
          } else {
            r.resume();
          }
        });

    page.navigate("https://selenium.dev");

    int height = (int) page.locator("img").first().evaluate("img => img.naturalHeight");
    Assertions.assertEquals(0, height);
  }
}

File: dotnet/Playwright/Request/AbortTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class AbortTest : TestBase
{
    [TestMethod]
    public async Task UnconditionalAbort()
    {
        await Page!.RouteAsync("**/*.css", route => route.AbortAsync());

        await Page.GotoAsync("https://selenium.dev");
        var height = (await Page.Locator("svg").First.BoundingBoxAsync())?.Height;
        Assert.IsTrue(height > 30);
    }

    [TestMethod]
    public async Task ConditionalAbort()
    {
        await Page!.RouteAsync("**/*", async route =>
        {
            if (route.Request.ResourceType == "image")
            {
                await route.AbortAsync();
            }
            else
            {
                await route.ContinueAsync();
            }
        });

        await Page.GotoAsync("https://selenium.dev");

        var height = await Page.Locator("img").First.EvaluateAsync<int>("img => img.naturalHeight");
        Assert.AreEqual(0, height);
    }
}